src/script/api/script_group.cpp

```diff
@@ -150,21 +150,27 @@
 	return ScriptObject::DoCommand(0, group_id << 16, (::INVALID_ENGINE << 16) | engine_id, CMD_SET_AUTOREPLACE);
 }
 
-/* static */ Money ScriptGroup::GetProfitThisYear(GroupID group_id)
+template <typename T, typename Func>
+T SumVehiclesInGroup(GroupID group_id, Func func)
```

Comment:
does this need to be a template parameter? Wouldn't `int func(Vehicle *)` type be sufficient?

Reply:
Certainly can't be an int, since it needs to work with the OverflowSafeInt* types and remain overflow safe.